### PR TITLE
[NTGDI:FREETYPE] Account for non-breaking spaces in x-dim of IntExtTextOutW

### PR DESCRIPTION
**Pull request by Doug-Lyons**
CORE-19768
Follow up of PR #7274. This stops WARN's from FF 28 as well on ASCII 0x007f (del characters).
Addendum to https://github.com/reactos/reactos/commit/a1bff5b94efa36b9e48d37f91b56428131c10531

## Purpose

_Account for x-dimension for non-breaking space characters._

JIRA issue: [CORE-19768](https://jira.reactos.org/browse/CORE-19768)

## Proposed changes

_Test for ASCII char 0x00a0 and account for typical space character in x-dimension._
_Ignore ASCII chars less that space (0x0020) since they are normally control characters._
_Ignore ASCII char 'del' (0x007f) because this is a 'DELETE' character and should have no width._

### CODE REVIEW

**Comment by SergeGautherie** [comment]
Please, have a look at  #7458.

**Review by tkreuzer — APPROVED** [review]

File: win32ss/gdi/ntgdi/freetype.c [hunk 1]
```diff
@@ -6795,7 +6795,8 @@ IntExtTextOutW(
     FONT_CACHE_ENTRY Cache;
     FT_Matrix mat;
     BOOL bNoTransform;
-    DWORD ch0, ch1, etx = 3; // etx is ASCII End of Text
+    DWORD ch0, ch1, nbsp = 0xa0; // nbsp is a non-breaking space
+    DWORD del = 0x7f; // DEL is ASCII DELETE
```
Comment (learn-more) [inline win32ss/gdi/ntgdi/freetype.c:6799]: Nit, it might make sense to make `del` and `nbsp` const?

**Review by learn-more — APPROVED** [review]

File: win32ss/gdi/ntgdi/freetype.c (hunk 1 above)
Comment (HBelusca) [reply win32ss/gdi/ntgdi/freetype.c:6799]: or as in Serge's PR, using #defines. (`#define ASCII_ETX 0x03`, `#define ASCII_DEL 0x7f`, etc.)

Comment (Doug-Lyons) [reply win32ss/gdi/ntgdi/freetype.c]: I changed them to ```const``` as Mark suggested. @HBelusca, using ````#define```'s would have been good as well. Thanks.

**Comment by Doug-Lyons** [comment]
> Please, have a look at #7458.

@SergeGautherie, I looked at your PR and it has its good points. I really wanted to keep the two functions separated.
One for testing for ```char``` values that are unusual and might need widths makes it easy to remove later.
This is likely once this code is run for a while and we collect data on what shows up here with the ```WARN:```.
Then the other that accounts for the actual widths stands on its own and can be edited to include more ```char``` values if needed.
Thank you for your PR, your suggestions, and comments.

**Review by DarkFire01 — APPROVED** [review]

**Review by HBelusca — APPROVED** [review]
Ok, but how does this code behave with other types of non breaking spaces etc? 
https://en.wikipedia.org/wiki/Non-breaking_space#Width_variation

**Comment by Doug-Lyons** [comment]
> Ok, but how does this code behave with other types of non breaking spaces, or with other zero-width characters, etc? https://en.wikipedia.org/wiki/Non-breaking_space#Width_variation https://ptiglobal.com/the-beauty-of-unicode-zero-width-characters/

@HBelusca, Thank you for providing these references. I believe that until we have ```real-world``` applications that actually make use of these features, we cannot provide programming to handle these adequately. The first question that comes to my mind, is "Do these characters show up in freetype as ```bitSize.cx == 0```. Once we know this and have example to test against, we should probably re-visit this PR and try and improve on it based on our findings. I propose to leave this issue as a future feature request or another PR. Thanks again.

**Comment by Doug-Lyons** [comment]
Thanks for everyone's help and approval on this.

**Pull request merged**